AppointEase/AppointEase.Application/Services/PatientService.cs

```diff
@@ -0,0 +1,153 @@
+﻿using AppointEase.Application.Contracts.Identity;
+using AppointEase.Application.Contracts.Interfaces;
+using AppointEase.Application.Contracts.Models.DbModels;
+using AppointEase.Application.Contracts.Models.Operations;
+using AppointEase.Application.Contracts.ModelsDto;
+using AppointEase.AspNetCore.Validator;
+using AutoMapper;
+using FluentValidation;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using OpenQA.Selenium;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AppointEase.Application.Services
+{
+    public class PatientService : IPatientService
+    {
+        private readonly IRepository<TblPacient> _uRepository;
+        private readonly UserManager<ApplicationUser> _userManager;
+        private readonly IMapper _mapper;
+        private readonly ICommon _common;
+        private readonly IValidator<PatientRequest> _personValidator;
+
+        public PatientService(IRepository<TblPacient> uRepository, UserManager<ApplicationUser> userManager, IMapper mapper, ICommon common, IValidator<PatientRequest> validator)
+        {
+            _uRepository = uRepository;
+            _userManager = userManager;
+            _mapper = mapper;
+            _common = common;
+            _personValidator = validator;
+        }
+
+        public async Task<OperationResult> CreatePersonAsync(PatientRequest patientRequest)
+        {
+            try
+            {
+                var patient = _mapper.Map<TblPacient>(patientRequest);
+                var patients = await GetAllPersons();
+
+                _personValidator.ValidateAndThrow(patientRequest);
```

Comment:
The method is `asynchronous`, why aren't we validating it `asynchronously`? We can use the `_validateAsync `method and check if it's valid with an if below, please check it.